plugin/trino-hive/src/test/java/io/trino/plugin/hive/fs/TestDirectoryListingCacheKey.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.fs;
+
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestDirectoryListingCacheKey
+{
+    @Test
+    public void testRecursiveFilesOnlyFlag()
+    {
+        Path path = new Path("file://some-path.txt");
+        assertTrue(new DirectoryListingCacheKey(path, true).isRecursiveFilesOnly());
+        assertFalse(new DirectoryListingCacheKey(path, false).isRecursiveFilesOnly());
+    }
```

Comment:
Pls add a test for `io.trino.plugin.hive.fs.DirectoryListingCacheKey#allKeysWithPath` as well.

Reply:
I just deleted this test altogether, it made more sense when the implementation for `isRecursiveFilesOnly()` was bit-twiddling the `hashCode` field in an attempt to reduce the object footprint, but now testing that a boolean is correctly assigned in the constructor seems pointless.